app/src/lib/dispatcher/error-handlers.ts

```diff
@@ -245,6 +246,74 @@ export async function pushNeedsPullHandler(
   return error
 }
 
+/**
+ * Handler for detecting when a merge conflict is reported to direct the user
+ * to a different dialog than the generic Git error dialog.
+ */
+export async function mergeConflictHandler(
+  error: Error,
+  dispatcher: Dispatcher
+): Promise<Error | null> {
+  const e = asErrorWithMetadata(error)
+  if (!e) {
+    return error
+  }
+
+  const gitError = asGitError(e.underlyingError)
+  if (!gitError) {
+    return error
+  }
+
+  const dugiteError = gitError.result.gitError
+  if (!dugiteError) {
+    return error
+  }
+
+  if (dugiteError !== DugiteError.MergeConflicts) {
+    return error
+  }
+
+  const { repository, gitContext } = e.metadata
+  if (repository == null) {
+    return error
+  }
+
+  if (!(repository instanceof Repository)) {
+    return error
+  }
+
+  if (gitContext == null) {
+    return error
+  }
+
+  switch (gitContext.kind) {
+    case 'pull':
+      dispatcher.mergeConflictDetectedFromPull()
+      break
+    case 'merge':
+      dispatcher.mergeConflictDetectedFromExplicitMerge()
+      break
+  }
+
+  if (gitContext.kind === 'merge') {
+    const { tip, branch } = gitContext
+    if (tip == null || tip.kind !== TipState.Valid) {
+      return error
+    }
+
+    dispatcher.showPopup({
+      type: PopupType.MergeConflicts,
+      repository,
+      currentBranch: tip.branch.name,
+      comparisonBranch: branch,
+    })
+  } else {
+    // TODO: what should we display if this is a pull?
```

Comment:
@outofambit previously this error handler would also fire when you triggered a conflict via `pull` (which is merging the remote tracking branch with your local branch). It looks like this dialog now expects this to be a "comparison branch" - will providing a remote branch name confuse your dialog? Could `comparisonBranch` be renamed to something more generic in Git (like `theirs`)?

Reply:
oh i see what you're saying. yeah that totally makes sense. perhaps `theirBranch`?

Reply:
(also a remote branch name won't confuse the dialog, that should be totally fine)

Reply:
Cool, let me push a commit to rename this and see how it feels.